aten/src/ATen/native/quantized/cpu/qnnpack/src/deconv-run.cc

```diff
@@ -129,20 +129,26 @@ enum pytorch_qnnp_status qnnpackDeConv(
   const size_t input_pixel_stride = deconv_p.input_channels;
   const size_t output_pixel_stride = deconv_p.output_channels;
 
-  pytorch_qnnp_status status = pytorch_qnnp_setup_deconvolution2d_nhwc_q8(
-      deconvolution,
-      batch_size,
-      input_height,
-      input_width,
-      input,
-      input_pixel_stride,
-      output,
-      output_pixel_stride,
-      threadpool);
-  if (status != pytorch_qnnp_status_success) {
-    pytorch_qnnp_log_error(
-        "failed to run decovolution op setup to setup indirection buffer.");
-    return status;
+  if (deconvolution->input != input ||
+      deconvolution->batch_size != batch_size ||
+      deconvolution->input_height != input_height ||
+      deconvolution->input_width != input_width ||
+      deconvolution->input_pixel_stride != input_pixel_stride) {
+    pytorch_qnnp_status status = pytorch_qnnp_setup_deconvolution2d_nhwc_q8(
+        deconvolution,
+        batch_size,
+        input_height,
+        input_width,
+        input,
+        input_pixel_stride,
+        output,
+        output_pixel_stride,
```

Comment:
I've got a feeling output and output_pixel_stride should also be part of the check.  What if input has not changed but output or output_pixel_stride have?  Looking at the implementation of `pytorch_qnnp_setup_deconvolution2d_nhwc_q8` it seems these two fields also change the internal state of the operator.

Reply:
The reason I did not include this is because while setup function does store it in operator, our integration of conv/deconv does not run via run_operator. Also output_pixel_stride is derived from conv_params_t which is setup once. So only thing that can potentially change is output pointer, but since we are not relying on that to run the operator, it seemed ok. 
As opposed to this in XNNPACK integration we do the check for output ptr as well since we cannot make any assumption on it.